### ISSUE TITLE
clippy-driver should have an option to get underlying rustc version

### ISSUE THREAD

**Issue by jsgf** [rust-lang/rust-clippy#5159]
At present, `clippy-driver --version` shows:
```
clippy 0.0.212 (c8e3cfb 2019-10-28)
```
which gives no clue to the underlying Rust version. `clippy-driver rustc --version` shows the same output.

A number of build scripts use the compiler version to determine what language/library features are available, typically by executing `$RUSTC --version` (either directly or via the `rustversion` crate). If `RUSTC=clippy-driver` then they'll get confused. (`cargo` does not set `RUSTC=clippy-driver` even when doing clippy builds, but arguably it should.)

I suggest one of:
1. `--version` shows the rustc version, and add `--clippy-version` shows the clippy version, or
2. `clippy-driver --version` shows the clippy version, but `clippy-driver rustc --version` shows the rustc version

**Comment by dtolnay** [comment]
> `clippy-driver --version` shows the clippy version, but `clippy-driver rustc --version` shows the rustc version

+1 for this option. This makes sense to me and we can accommodate it in the rustversion crate.

**Comment by matthiaskrgr** [comment]
````
clippy-driver -Vv

rustc 1.43.0-nightly (58b834344 2020-02-05)
binary: rustc
commit-hash: 58b834344fc7b9185e7a50db1ff24e5eb07dae5e
commit-date: 2020-02-05
host: x86_64-unknown-linux-gnu
release: 1.43.0-nightly
LLVM version: 9.0
````

**Comment by jsgf** [comment]
Ah, I thought I'd tried that but I guess not.

**Comment by dtolnay** [comment]
Thanks @matthiaskrgr, rustversion can use that for now. I think it would still be good to have `clippy-driver rustc --version` print the rustc version instead of the clippy version.

**Comment by flip1995** [comment]
@dtolnay `clippy-driver --rustc` now forwards everything to the underlying `rustc`. So `clippy-driver --rustc --version`¹ will print the same as calling the underlying `rustc --version` would.

This is already included in the Clippy version shipped with `beta-1.46`.

---

¹: We couldn't use `clippy-driver rustc --version`, because when clippy is used with cargo `clippy-driver rustc <--some-args>` is called by cargo.
